internal/content_coding_test.go

```diff
@@ -113,3 +156,168 @@ func TestStreamGzipDecode(t *testing.T) {
 
 	require.Equal(t, []byte("howdy"), b[:n])
 }
+
+func BenchmarkZstdEncodeDecode(b *testing.B) {
+	enc, err := NewZstdEncoder(1)
+	require.NoError(b, err)
+	dec, err := NewZstdDecoder()
+	require.NoError(b, err)
+
+	for n := 0; n < b.N; n++ {
+		payload, err := enc.Encode([]byte("howdy"))
+		require.NoError(b, err)
+
+		actual, err := dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "howdy", string(actual))
+	}
+}
+
+func BenchmarkZstdReuse(b *testing.B) {
+	enc, err := NewZstdEncoder(1)
+	require.NoError(b, err)
+	dec, err := NewZstdDecoder()
+	require.NoError(b, err)
+
+	for n := 0; n < b.N; n++ {
+		payload, err := enc.Encode([]byte("howdy"))
+		require.NoError(b, err)
+
+		actual, err := dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "howdy", string(actual))
+
+		payload, err = enc.Encode([]byte("doody"))
+		require.NoError(b, err)
+
+		actual, err = dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "doody", string(actual))
+	}
+}
+
+func BenchmarkGzipEncodeDecode(b *testing.B) {
+	enc, err := NewGzipEncoder(-1)
+	require.NoError(b, err)
+	dec := NewGzipDecoder()
+
+	for n := 0; n < b.N; n++ {
+		payload, err := enc.Encode([]byte("howdy"))
+		require.NoError(b, err)
+
+		actual, err := dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "howdy", string(actual))
+	}
+}
+
+func BenchmarkGzipReuse(b *testing.B) {
+	enc, err := NewGzipEncoder(-1)
+	require.NoError(b, err)
+	dec := NewGzipDecoder()
+
+	for n := 0; n < b.N; n++ {
+		payload, err := enc.Encode([]byte("howdy"))
+		require.NoError(b, err)
+
+		actual, err := dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "howdy", string(actual))
+
+		payload, err = enc.Encode([]byte("doody"))
+		require.NoError(b, err)
+
+		actual, err = dec.Decode(payload, maxDecompressionSize)
+		require.NoError(b, err)
+
+		require.Equal(b, "doody", string(actual))
+	}
+}
```

Comment:
You don't need this one.

Reply:
Why though? Why having the ZstdReuse and removing this?

Reply:
You don't need it if you change the benchmark above as it will cover the reuse case already. :-) What I mean here is remove all the reuse tests and do it in the normal test. I don't think there is a point in measuring the allocation speed of the machine...